### DISCUSSION TITLE
Guideline on what kind of computations are achievable on actual Loihi hardware

### DISCUSSION THREAD

**Discussion by CloudyDory** [lava-nc/lava#524] [Q&A]
Hi, we are new INRC members (and new to LAVA and Loihi). We have some previously trained SNN models (by PyTorch and runs on CPU/GPU) and hope to convert the models to LAVA and run on Loihi. However, I am a bit unsure about what kind of computations are achievable on actual Loihi hardware, for example some of my current worrying points are:
  (1). In the first layer of our SNN, we choose to directly convolve on raw image values and then pass the results to spiking neurons, whereas in the LAVA tutorial the images are first converted to spike trains before entering the first layer of SNN. Is it mandatory to use spikes as input, or can we use raw image pixel values as constant input current into Loihi neurons?
  (2). Some of our SNN models have a layer that calculates weighted average (weights are predefined and fixed) of input spikes from all input neurons at every time step, and outputs floating point numbers to the next layer of SNN. Can the Loihi chip run this layer?

In general, is there a guideline on what kinds of computations are achievable on actual Loihi hardware, and what kinds are not? If a SNN model can run on CPU by the LAVA Loihi simulator, does it mean that the model can also run on actual Loihi hardware (assuming that the model is not too big to fit on the chip)?

**Comment by EternalGoldenBraid** [comment]
Hey,

I have similar questions. Did you by any chance find answers to these questions? Particularily in the simulator-cpu and Loihi correspondence?

**Comment by Deng-Xian-Sheng** [comment]
Practice is the only way to test truth

**Reply by EternalGoldenBraid** [reply]
Beginners mind is the way
